### PR TITLE
feat: add a cache control headers to the jwks.json endpoint

### PR DESCRIPTION
**Pull request by porcellus**
## Summary of change

add a cache control headers to the jwks.json endpoint

## Related issues

-   https://github.com/supertokens/supertokens-node/issues/617

## Test Plan

- Added test cases that check the new header

## Documentation changes

- [ ] Requires updating the [NextJS Setup guide](https://supertokens.com/docs/thirdpartyemailpassword/nextjs/setting-up-backend)

## Checklist for important updates

-   [x] Changelog has been updated
-   [x] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [x] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [x] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [x] If new thirdparty provider is added,
    -   [ ] update switch statement in `recipe/thirdparty/providers/configUtils.ts` file, `createProvider` function.
    -   [ ] add an icon on the user management dashboard.
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [x] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [x] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [x] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

### CODE REVIEW

File: examples/next/with-emailpassword/pages/api/auth/[[...path]].tsx [hunk 1]
```diff
@@ -10,7 +10,10 @@ export default async function superTokens(req: NextApiRequest, res: NextApiRespo
     await superTokensNextWrapper(
         async (next) => {
             // This is needed for production deployments with Vercel
-            res.setHeader("Cache-Control", "no-cache, no-store, max-age=0, must-revalidate");
+            if (!response.hasHeader("Cache-Control")) {
```
Comment (rishabhpoddar) [inline examples/next/with-emailpassword/pages/api/auth/[[...path]].tsx:13]:
```suggestion
            if (!res.hasHeader("Cache-Control")) {
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: examples/next/with-emailpassword/config/frontendConfig.tsx [hunk 2]
```diff
@@ -8,7 +8,12 @@ export let frontendConfig = () => {
         appInfo,
         // recipeList contains all the modules that you want to
         // use from SuperTokens. See the full list here: https://supertokens.com/docs/guides
-        recipeList: [EmailPasswordReact.init(), SessionReact.init()],
+        recipeList: [
+            EmailPasswordReact.init({
+                useShadowDom: false,
+            }),
```
Comment (rishabhpoddar) [inline examples/next/with-emailpassword/config/frontendConfig.tsx:14]: why is this needed?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: examples/next/with-emailpassword/config/frontendConfig.tsx (hunk 2 above)
Comment (porcellus) [reply examples/next/with-emailpassword/config/frontendConfig.tsx:14]: I've removed it, but the app was failing to render in "dev mode".
